controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/kvtable/CreateTableTask.java

```diff
@@ -0,0 +1,122 @@
+        .thenCompose(id -> {
+            if (!id.equals(kvTableId)) {
+                // we don;t execute the request if Table IDs do not match
+                return CompletableFuture.completedFuture(null);
+            } else {
+                this.kvtMetadataStore.createKeyValueTable(scope, kvt, config, creationTime, null, executor)
+                        .thenComposeAsync(response -> {
+                            // only if its a new kvtable or an already existing non-active kvtable then we will create
+                            // segments and change the state of the kvtable to active.
+                            if (response.getStatus().equals(CreateKVTableResponse.CreateStatus.NEW) ||
+                                    response.getStatus().equals(CreateStreamResponse.CreateStatus.EXISTS_CREATING)) {
+                                final int startingSegmentNumber = response.getStartingSegmentNumber();
+                                final int minNumSegments = response.getConfiguration().getPartitionCount();
+                                List<Long> newSegments = IntStream.range(startingSegmentNumber, startingSegmentNumber + minNumSegments)
+                                        .boxed()
+                                        .map(x -> NameUtils.computeSegmentId(x, 0))
+                                        .collect(Collectors.toList());
+                                return kvtMetadataTasks.notifyNewSegments(scope, kvt, newSegments, requestId)
+                                        .thenCompose(y -> {
+                                            final KVTOperationContext context = kvtMetadataStore.createContext(scope, kvt);
+                                            //TODO: add withRetries
+                                            kvtMetadataStore.getVersionedState(scope, kvt, context, executor)
+                                                    .thenCompose(state -> {
+                                                        if (state.getObject().equals(State.CREATING)) {
+                                                            kvtMetadataStore.updateVersionedState(scope, kvt, KVTableState.ACTIVE,
+                                                                    state, context, executor);
+                                                        }
+                                                        return CompletableFuture.completedFuture(null);
+                                                    });
+                                            return CompletableFuture.completedFuture(null);
+                                        });
+                            }
+                            return CompletableFuture.completedFuture(null);
+                        });
+                return CompletableFuture.completedFuture(null);
+             }
+        });
+    }
+
+    private Controller.CreateKeyValueTableStatus.Status translate(CreateKVTableResponse.CreateStatus status) {
```

Comment:
this response is not being sent by the task.. the task completes asynchronously.. 
The submitter of the task doesnt get the response from the task.. it checks for task completion condition (desired goal state).

Reply:
Ok. Will remove unused method.

Reply:
done.